### PR TITLE
Adopt typed throws in one withUnsafePointer() API

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by glessard — APPROVED** [review]

File: stdlib/public/core/LifetimeManager.swift [hunk 1]
```diff
@@ -120,25 +119,27 @@ public func _withUnprotectedUnsafeMutablePointer<T, Result>(
...
+) throws(E) -> Result
 {
   return try body(UnsafePointer<T>(Builtin.addressOfBorrow(value)))
 }
-#else
-// TODO: This should be unified with non-embedded Swift.
-@inlinable
-public func withUnsafePointer<T, E, Result>(
+
+/// ABI: Historical withUnsafePointer(to:_:) rethrows, expressed as "throws",
```
Comment (DougGregor) [inline stdlib/public/core/LifetimeManager.swift:132]: Should this be in a `#if !$Embedded` block, since there is no ABI to maintain?

**Review by DougGregor — APPROVED** [review]
LGTM!

File: stdlib/public/core/LifetimeManager.swift (hunk 1 above)
Comment (stephentyrone) [reply stdlib/public/core/LifetimeManager.swift:132]: I wonder if we shouldn't put all these legacy untyped-throws implementations in a file that we can simply not built for non-ABI or new platforms.

Comment (kubamracek) [reply stdlib/public/core/LifetimeManager.swift:132]: I'll leave this is it is in *this* PR, if you're okay with that, but yes, we want to decide how to organize the legacy implementations. I don't feel strongly about this, but let me just note that there is no impact for Embedded Swift even if we keep the legacy impl in -- unused functions do not contribute to binary outputs.

**Comment by kubamracek** [comment]
@swift-ci please test

**Review by stephentyrone — APPROVED** [review]

**Pull request merged**